src/elements/emby-progressring/emby-progressring.js

```diff
@@ -11,23 +11,22 @@ define(['require', 'css!./emby-progressring', 'webcomponents'], function (requir
         require(['text!./emby-progressring.template.html'], function (template) {
             instance.innerHTML = template;
 
-            //if (window.MutationObserver) {
-            //    // create an observer instance
-            //    var observer = new MutationObserver(function (mutations) {
-            //        mutations.forEach(function (mutation) {
-
-            //            instance.setProgress(parseFloat(instance.getAttribute('data-progress') || '0'));
-            //        });
-            //    });
-
-            //    // configuration of the observer:
-            //    var config = { attributes: true, childList: false, characterData: false };
-
-            //    // pass in the target node, as well as the observer options
-            //    observer.observe(instance, config);
-
-            //    instance.observer = observer;
-            //}
+            if (window.MutationObserver) {
+                // create an observer instance
+                var observer = new MutationObserver(function (mutations) {
+                    mutations.forEach(function (mutation) {
+                        instance.setProgress(parseFloat(instance.getAttribute('data-progress') || '0'));
+                    });
+                });
+
+                // configuration of the observer:
+                var config = { attributes: true, childList: false, characterData: false };
+
+                // pass in the target node, as well as the observer options
+                observer.observe(instance, config);
+
+                instance.observer = observer;
+            }
```

Comment:
Looks like we don't set `data-progress` anywhere.

Reply:
https://github.com/jellyfin/jellyfin-web/blob/master/src/controllers/dashboard/mediaLibrary.js#L282

https://github.com/jellyfin/jellyfin-web/blob/master/src/components/cardbuilder/cardBuilder.js#L1417

This would essentially allow us to dynamically update the progress on library refreshes, if we wish to.

Reply:
Currently, this blocks updating (by function?) and I see `0%`.
To reproduce, refresh one library.

Reply:
I changed the way progress is updated in emby-itemrefreshindicator to use mutation instead of using setProgress directly in 44174a5, which fixes progress when refreshing libraries for me.

Reply:
The main benefit of MutationObserver is that it's asynchronous. So we don't have to manually block execution each time something happens to update it. We simply set the value and let the JS engine determine when the best time to process the change is (It's a microtask, so it gets done once the queue is empty).

It doesn't use events and, managed properly, shouldn't really impact performance.